src/main/java/InfrastructureManager/Rest/RestRunner.java

```diff
@@ -0,0 +1,111 @@
+package InfrastructureManager.Rest;
+
+import InfrastructureManager.*;
+import spark.Spark;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+
+public class RestRunner extends Runner {
+    private int port = 4567;
+    private int retryAttempt = 10;
+    private String host = "localhost";
+    private String heartbeatPath = "/heartbeat";
+    private static RestRouter server = null;
+    private static RestRunner restRunner = null;
+
+    public RestRunner(String name, MasterInput input, MasterOutput...outputs) {
+        super(name,input,outputs);
+    }
+
+    private void startServer() {
+        this.server = RestRouter.startRouter(port);
+    }
+
+    /**
+     * Starter for the server if it's not started yet, the system would wait up to 10s for the server to start
+     * Made public for Testing purpose
+     */
+    public void startServerIfNotRunning() throws InterruptedException {
+        try {
+            if(!serverIsRunning()) {
+                startServer();
+            }
+        } catch (IllegalStateException e) {
+            throw e;
+        }
+
+        waitForServerToRun();
+    }
+
+    private void waitForServerToRun() throws InterruptedException {
+        int tries = retryAttempt;
+        while(tries > 0) {
+            if(!serverIsRunning()) {
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    throw e;
+                }
+            } else {
+                return;
+            }
+            tries--;
+        }
+    }
+
+    private boolean serverIsRunning(){
+        try{
+            HttpURLConnection con = (HttpURLConnection)new URL("http",host, port, heartbeatPath).openConnection();
+            return con.getResponseCode()==200;
+        }catch(Exception e){
+            return false;
+        }
+    }
+
+    /**
+     * Method for stoping the static Spark instance and sets server to null after Spark is stopped
+     */
+    public void killServer() {
+        this.server = null;
+        Spark.stop();
+        Spark.awaitStop();
+    }
+
+    /**
+     * Method for getting the Singleton REST Runner instance or set from Configuration if not available
+     */
+    public static RestRunner getRestRunner() {
+        if(restRunner == null) {
+            restRunner = (RestRunner) Master.getInstance().getRunnerConfiguration("Rest");
+        }
+        return restRunner;
+    }
+
+    /**
+     * Method for getting the Singleton REST Runner instance or set from parameter if not available
+     */
+    public static RestRunner getRestRunner(Runner runner) {
+        if(restRunner == null) {
+            restRunner = (RestRunner) runner;
+        }
+        return restRunner;
+    }
```

Comment:
There is no need to interface with the master here. Just return a value using the private constructor. 

```suggestion
  public static RestRunner getRestRunner(String name, MasterInput input, MasterOutput...outputs) {
        if (restRunner == null) {
            restRunner = new RestRunner(String name, MasterInput input, MasterOutput...outputs);
        }
        return restRunner;
    }

    public static RestRunner getRestRunner() {
        return getRestRunner("",null,null);
    }
```